### PR TITLE
⬆️ Allow SQLAlchemy v1.4

### PR DESCRIPTION
**Pull request by stevenheidel**
This version constraint was preventing us from installing both SQLAlchemy 1.4 and jupyter-book in the same environment.

### CODE REVIEW

**Comment by jaapz** [comment]
@chrisjsewell please review this and merge, projects using sqlalchemy's latest stable release are currently unable to use myst_nb (if using an up to date version of pip)

**Comment by choldgraf** [comment]
Hmmmm - in general our pinning strategy is to pin to minor versions and wildcard on patches (thus, the `~=`). This PR doesn't just update the version, it also changes the pinning policy. As long as the tests are passing, it seems like supporting either 1.3.* or 1.4.* would be OK, since those are both well-used versions (I believe 1.4.* is the most in-development and is likely to be more similar to 2.0).

That said, I'm not sure why the tests aren't running here, I'll try and update this PR and see if that triggers them

File: setup.py [hunk 1]
```diff
@@ -41,7 +41,7 @@
         # note: we do not yet pin to higher, since there are reports of issues with 0.3
         # see: https://github.com/jupyter/nbclient/issues/58
         "nbclient>=0.2,<0.6",
-        "sqlalchemy~=1.3.12",
+        "sqlalchemy>=1.3.12",
```
Comment (choldgraf) [inline setup.py:44]:
```suggestion
        "sqlalchemy~=1.3.12,~=1.4.22",
```

File: setup.py [hunk 2]
```diff
@@ -41,7 +41,7 @@
         # note: we do not yet pin to higher, since there are reports of issues with 0.3
         # see: https://github.com/jupyter/nbclient/issues/58
         "nbclient>=0.2,<0.6",
-        "sqlalchemy~=1.3.12",
+        "sqlalchemy~=1.3,<1.5",
```
Comment (choldgraf) [inline setup.py:44]:
```suggestion
        "sqlalchemy>=1.3.12,<1.5",
```

**Pull request merged**